### DISCUSSION TITLE
Exposing public / aliased objects?

### DISCUSSION THREAD

**Discussion by florianmartens** [prisma/prisma#7157] [Q&A] [answered]
Hi,

As I'm learning Prisma a couple of questions arise. One that pops of frequently is object APIs. Let's say I have a User schema defined:
```prisma
model User {
  id        String    @id @default(cuid())
  email     String    @unique
  password  String
  salt      String

  @@map(name: "users")
}
```

When including the user in a query, I obviously don't want to send the password (even though it's hashed) as part of the JSON response. I'd love to have a way to define a PublicUser that I can reuse in different parts of my app. Of course, I can do:

```ts
include {
  user: {
     email: true
     ...
  }
}
```
But it's easy to forget this or be inconsistent with the fields.

I'd love to know how others are handling this as I'm sure it comes up often.

**Comment by ryands17** [comment, marked as answer]
@florianmartens 👋 

We do have a requests for this here that you can follow:

https://github.com/prisma/prisma/issues/3796
https://github.com/prisma/prisma/issues/5042

One workaround for now would be to create a helper function that does this:

```ts
function userSelect(params?: Prisma.UserSelect): Prisma.UserSelect {
  return {
    ...params,
    id: true,
    email: true,
    salt: false,
    password: false,
  }
}

await prisma.user.findMany({ select: { ...userSelect() } })
```
